stubs/psutil/psutil/__init__.pyi

```diff
@@ -241,16 +241,27 @@ def wait_procs(
     procs: Iterable[Process], timeout: float | None = None, callback: Callable[[Process], object] | None = None
 ) -> tuple[list[Process], list[Process]]: ...
 def cpu_count(logical: bool = True) -> int: ...
-def cpu_times(percpu: bool = False): ...
+@overload
+def cpu_freq(percpu: Literal[False] = ...) -> scpufreq: ...
+@overload
+def cpu_freq(percpu: Literal[True]) -> list[scpufreq]: ...
+@overload
+def cpu_times(percpu: Literal[False] = ...) -> pcputimes: ...
```

Comment:
This doesn't seem correct. According to stubs, the returned fields are:

```python
class pcputimes(NamedTuple):
    user: float
    system: float
    children_user: float
    children_system: float
```

On my linux system, I get no `children_user` or `children_system` fields, and the type is `scputimes`, not `pcputimes`:

```python
>>> psutil.cpu_times()
scputimes(user=521.89, nice=0.0, system=116.9, idle=5597.83, iowait=209.84, irq=0.0, softirq=11.94, steal=0.0, guest=0.0, guest_nice=0.0)
>>> type(psutil.cpu_times())
<class 'psutil._pslinux.scputimes'>
```

Currently `scputimes` is defined in various places, so presumably you need to import it from the right place depending on platform:

```
akuli@akuli-desktop:~/typeshed/stubs/psutil$ git grep scputimes
psutil/_psaix.pyi:class scputimes(NamedTuple):
psutil/_psbsd.pyi:class scputimes(NamedTuple):
psutil/_pslinux.pyi:def set_scputimes_ntuple(procfs_path) -> None: ...
psutil/_pslinux.pyi:scputimes: Any
psutil/_psosx.pyi:class scputimes(NamedTuple):
psutil/_pssunos.pyi:class scputimes(NamedTuple):
psutil/_pswindows.pyi:class scputimes(NamedTuple):
```

The `scputimes: Any` looks wrong. It is generated dynamically at runtime and needs better stubs. (Just include all possible fields and perhaps add a comment.)

Alternatively, if you don't want to bother with `scputimes`, you could just use `Incomplete` instead of `pcputimes` in all these methods for now.

Reply:
You're right -- on my machine it was a `psutil._pslinux.scputimes` as well. It looks like `pcputimes` is returned for `psutil.Process(...).cpu_times()`.